### ISSUE TITLE
ClassCastException on UnbindView

### ISSUE THREAD

**Issue by ncapdevi** [mikepenz/FastAdapter#268]
Class **A** descends from AbstractItem. Class **B** descends from class **A**. Class **B.ViewHolder**  extends from **A.ViewHolder**.  I get this exception

```
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                   Process: mypackge, PID: 24471
                                                                   java.lang.ClassCastException: mypackge.views.A.Viewholdercannot be cast to mypackge.views.B.Viewholder
                                                                       at mypackge.views.B.unbindView(B.java:35)(mypackge.views.B.java:35)
                                                                       at com.mikepenz.fastadapter.FastAdapter$OnBindViewHolderListenerImpl.unBindViewHolder(FastAdapter.java:1847)
                                                                       at com.mikepenz.fastadapter.FastAdapter.onViewRecycled(FastAdapter.java:582)
                                                                       at android.support.v7.widget.RecyclerView$Recycler.dispatchViewRecycled(RecyclerView.java:5731)
                                                                       at android.support.v7.widget.RecyclerView$Recycler.addViewHolderToRecycledViewPool(RecyclerView.java:5496)
                                                                       at android.support.v7.widget.RecyclerView$Recycler.recycleViewHolderInternal(RecyclerView.java:5479)
                                                                       at android.support.v7.widget.RecyclerView.removeAnimatingView(RecyclerView.java:1242)
                                                                       at android.support.v7.widget.RecyclerView$ItemAnimatorRestoreListener.onAnimationFinished(RecyclerView.java:11156)
                                                                       at android.support.v7.widget.RecyclerView$ItemAnimator.dispatchAnimationFinished(RecyclerView.java:11656)
                                                                       at android.support.v7.widget.SimpleItemAnimator.dispatchRemoveFinished(SimpleItemAnimator.java:265)
                                                                       at com.mikepenz.itemanimators.BaseItemAnimator$4.onAnimationEnd(BaseItemAnimator.java:254)
                                                                       at android.support.v4.view.ViewPropertyAnimatorCompatJB$1.onAnimationEnd(ViewPropertyAnimatorCompatJB.java:47)
                                                                       at android.view.ViewPropertyAnimator$AnimatorEventListener.onAnimationEnd(ViewPropertyAnimator.java:1121)
                                                                       at android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1153)
                                                                       at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1313)
                                                                       at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
                                                                       at android.animation.AnimationHandler.-wrap2(AnimationHandler.java)
                                                                       at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
                                                                       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:872)
                                                                       at android.view.Choreographer.doCallbacks(Choreographer.java:686)
                                                                       at android.view.Choreographer.doFrame(Choreographer.java:618)
                                                                       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
                                                                       at android.os.Handler.handleCallback(Handler.java:751)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                       at android.os.Looper.loop(Looper.java:154)
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)

```

Meaning that it is trying to cast class **A.ViewHolder** to **B.Viewholder** on unBindView even though the correct viewholder type is being declared and has always worked for BindView.  I've tried implementing a ViewHolderFactory in class B and it didn't work either. Any ideas?

**Comment by mikepenz** [comment]
@ncapdevi I have found the issue already. And it is fixed in v2.1 (currently available as snapshot)
You can see the changes which were done to fix this here:
https://github.com/mikepenz/FastAdapter/commit/a6a3f2e8a8cf73a019a51147630c78fe82e66c6c

**Comment by ncapdevi** [comment]
Perfect. Thanks!
